src/caustics/__init__.py

```diff
@@ -1,16 +1,23 @@
 from ._version import version as VERSION  # noqa
 
-from .constants import *
-from .lenses import *
-from .cosmology import *
-from .packed import *
-from .parametrized import *
-from .light import *
-from .utils import *
-from .sims import *
-from .tests import *
+from . import constants, lenses, cosmology, packed, parametrized, light, utils, sims
+from .tests import test
 
 # from .demo import *
 
 __version__ = VERSION
 __author__ = "Ciela"
+
+__all__ = [
```

Comment:
Is the `__all__` needed when it references everything? Also, will `__version__` and `__author__` still be accessible if they are not in `__all__`?

Reply:
This exposes the modules that is public. For example, as you may have notice that originally, `data` module is not imported, so users wouldn't be able to just do `import caustics; caustics.data.HDF5Dataset`. If `data` was imported here and added to `__all__` then you would be able to do that.

At the end of the day, really this part of the thought process of what your final API is like. If it's expected that users can access modules straight from `caustics` imports, then this is what needs to happen. If a user is expected to import the functions from each individual module, then you don't need to specify it, so in the case of `data` since it's not in `__all__` users would have to do like `from caustics.data import HDF5Dataset`.